test/client_test.rb

```diff
@@ -0,0 +1,30 @@
+require "test_helper"
+require 'prometheus_exporter/client'
+
+# Mock queue reduces its length as it is checked
+class MockQueue
+  def length
+    @queue_length = (@queue_length || 3) - 1
+    @queue_length < 0 ? 0 : @queue_length
+  end
+end
+
+describe PrometheusExporter::Client do
+  
+  before do
+    @client = PrometheusExporter::Client.default
+  end
+
+  describe "flush" do
+    it "sends all queued entries within the timeout" do
+      @client.instance_variable_set(:@queue, MockQueue.new())
+      assert @client.flush(0.01)
```

Comment:
This is potentially a heisentest, cause sleep 0.001 is not a guaranteed to take 1ms.

Also ... ` sleep 0.001` is an implementation detail, no need to test that. 

Instead.... I would do an "honest" flush in a test, pause client, queue a bunch of messages, unpause/flush, check it is empty.

Reply:
Can do.
Do you prefer a series of discreet PR (1 for flush, 1 for the info).  That way you can pick/choose?
Thanks